tests/validation/tests/v3_api/test_cluster_templates.py

```diff
@@ -714,6 +716,224 @@ def test_cluster_template_create_with_monitoring():
+
+    cluster_cleanup(standard_user_client, cluster)
+
+
+def test_clustertemplate_readonly_member_edit_delete():
+
+    # As an admin, create cluster template/revision and provide "user/read-only
+    #  access" to standard user. The user should not be able to edit/delete
+    # cluster template/revision
+
+    k8sversionlist = get_k8s_versionlist()
+    admin_client = get_admin_client()
+    cluster_config = get_cluster_config(k8sversionlist[0])
+
+    # Obtain the principal ID of the standard user
+    principalId = user_token["standard_user"]["user"]["principalIds"]
+
+    # Create a cluster template and provide standard user "user access" to the
+    # cluster template
+    members = [{"type": "member", "accessType": "read-only",
+                "userPrincipalId": principalId}]
+
+    cluster_template_name = random_test_name("template")
+    cluster_template = \
+        admin_client.create_cluster_template(name=cluster_template_name,
+                                             description="test-template",
+                                             members=members)
+    clusterTemplateId = cluster_template.id
+    # Create Cluster template revision
+    revision_name = random_test_name("revision")
+    cluster_template_revision = \
+        admin_client.create_cluster_template_revision(
+            name=revision_name,
+            clusterConfig=cluster_config,
+            clusterTemplateId=clusterTemplateId,
+            enabled="true")
+
+    standard_user_client = \
+        get_client_for_token(
+            user_token["standard_user"]["token"])
+
+    # Verify the standard user does not have permission to edit/update the
+    # template as he has only user access
+    members = [{"type": "member", "accessType": "read-only",
+                "userPrincipalId": principalId},
+               {"type": "member", "accessType": "read-only",
+                "groupPrincipalId": "*"}]
+    with pytest.raises(ApiError) as e:
+        standard_user_client.update(cluster_template,
+                                    name="sjtest",
+                                    description="test-template",
+                                    members=members)
+        assert e.value.error.status == 403
+        assert e.value.error.code == "PermissionDenied"
+
+    # Verify the standard user does not have permission to delete the
+    # revision as he has only user/read-only access
+    with pytest.raises(ApiError) as e:
+        standard_user_client.delete(cluster_template_revision)
+        assert e.value.error.status == 403
+        assert e.value.error.code == "PermissionDenied"
+
+    # Verify the standard user does not have permission to delete the
+    # template as he has only user access
+    with pytest.raises(ApiError) as e:
+        standard_user_client.delete(cluster_template)
+        assert e.value.error.status == 403
+        assert e.value.error.code == "PermissionDenied"
+
+
+def validate_cluster_with_template(client, cluster,
```

Comment:
Is there a way to check the cluster template revision used by the cluster?

Reply:
I have added that check in the test method.

Reply:
This method is called `validate_cluster_with_template` but it does not validate anything about the template. 
I feel like the name will make more sense if we include the following two checks in this method. 
```
    check_cluster_version(cluster, k8sversionlist[1])
    assert cluster.clusterTemplateRevisionId == cluster_template_revision2.id
```